### PR TITLE
fix: Uniqueness and non-emptiness validation of column or tag names

### PR DESCRIPTION
**Pull request by bgzhao66**

### CODE REVIEW

File: src/client/src/tscSQLParser.c [hunk 1]
```diff
@@ -1628,7 +1628,7 @@ static bool validateTagParams(SArray* pTagsList, SArray* pFieldList, SSqlCmd* pC
       return false;
     }
 
-    if (has(pTagsList, i + 1, p->name) == true) {
+    if (has(pTagsList, i, p->name) == true) {
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:1631]: Check p->name against the slice [0, i) of pTagsList with i exclusive.

File: src/client/src/tscSQLParser.c [hunk 2]
```diff
@@ -1569,7 +1569,7 @@ static bool validateTableColumnInfo(SArray* pFieldList, SSqlCmd* pCmd) {
     }
 
     // field name must be unique
-    if (has(pFieldList, i + 1, pField->name) == true) {
+    if (has(pFieldList, i, pField->name) == true) {
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:1572]: Check name against the slice [0, i) of pFieldList with i exclusive.

File: src/client/src/tscSQLParser.c [hunk 3]
```diff
@@ -1657,8 +1657,8 @@ static bool validateTagParams(SArray* pTagsList, SArray* pFieldList, SSqlCmd* pC
   // field name must be unique
   for (int32_t i = 0; i < numOfTags; ++i) {
     TAOS_FIELD* p = taosArrayGet(pTagsList, i);
-
-    if (has(pFieldList, 0, p->name) == true) {
+    size_t      numOfCols = taosArrayGetSize(pFieldList);
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:1660]: Check name against the slice [0, numOfCol) of pFieldList with i exclusive.

File: src/client/src/tscSQLParser.c [hunk 4]
```diff
@@ -7552,7 +7551,9 @@ int32_t validateColumnName(char* name) {
     return validateColumnName(token.z);
   } else if (token.type == TK_ID) {
     stringProcess(name, token.n);
-    return TSDB_CODE_SUCCESS;
+    if (strlen(name) == 0) {
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:7554]: Check non-emptiness of name after stringProcess

**Review by bgzhao66 — COMMENTED** [review]
Post comments

File: src/client/src/tscSQLParser.c [hunk 5]
```diff
@@ -1804,9 +1804,8 @@ int32_t validateOneColumn(SSqlCmd* pCmd, TAOS_FIELD* pColField) {
 }
 
 /* is contained in pFieldList or not */
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:1806]: Check existence of a name against the part of pFieldlist that have already been processed.

File: src/client/src/tscSQLParser.c [hunk 6]
```diff
@@ -1657,8 +1657,8 @@ static bool validateTagParams(SArray* pTagsList, SArray* pFieldList, SSqlCmd* pC
   // field name must be unique
   for (int32_t i = 0; i < numOfTags; ++i) {
     TAOS_FIELD* p = taosArrayGet(pTagsList, i);
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:1659]: Check name against the slice [0, numOfCol) of pFieldList with numOfCols exclusive.

File: src/client/src/tscSQLParser.c [hunk 7]
```diff
@@ -7552,7 +7551,9 @@ int32_t validateColumnName(char* name) {
     return validateColumnName(token.z);
   } else if (token.type == TK_ID) {
     stringProcess(name, token.n);
```
Comment (bgzhao66) [inline src/client/src/tscSQLParser.c:7553]: Check non-emptiness of name after stringProcess

**Review by stephenkgu — APPROVED** [review]

**Pull request merged**